File: GoLang/5 kyu/Primes with even digits/main.go
package main

import (
	"math"
	"fmt"
)

func main() {
	fmt.Println(F(1000))
}

func isPrime(n int) bool {
  if n <= 1 {
    return false
  }
  if (n == 2 || n == 3) {
    return true
  }
  if (n % 2 == 0 || n % 3 == 0) {
    return false
  }
  
  for i := 5; i <= int(math.Sqrt(float64(n))); i += 6 {
    if (n % i == 0 || n % (i + 2) == 0) {
      return  false
    }
  }
  
  return true
}

func countEvens(n int) (count int) {
  for n > 0 {
    if (n%10) % 2 == 0 {
      count++
    }
    n /= 10
  }
  return count
}

func countNums(n int) (count int) {
  for n > 9 {
    count++
    n /= 10
  }
  count++
  if n == 1 {
    count--
  }
  return count
}

func F(n int) int {
  for i := n - 1; i > 0; i-- {
    if isPrime(i) && countEvens(i) == (countNums(i) - 1) {
    return i
    }
  }
  return 0
}

File: Python/5 kyu/Primes with even digits/main.py
from math import sqrt

def isPrime(num):
    if num <= 1: return False
    if num == 2 or num == 3: return True
    if num%2 == 0 or num%3 == 0: return False
    for i in range(5, int(sqrt(num) + 1), 6):
        if (num % i == 0) or (num % (i + 2) == 0): return False
    return True 

def countEvens(n):
    count = 0
    while n > 0:
        if (n%10) % 2 == 0:
            count += 1
        n //= 10
    return count

def countNums(n):
    count = 0
    while n > 9:
        count += 1
        n //= 10
    count += 1
    if n == 1:
        count -= 1
    return count
    

def f(n):
    for i in range(n - 1, 0, -1):
        if isPrime(i) and (countEvens(i) == (countNums(i) - 1)):
            return i
    return 0

if __name__ == "__main__":
    print(f(2856))